observer/mapCounters/mapCountersHolder.go

```diff
@@ -0,0 +1,63 @@
+package mapCounters
+
+import (
+	"errors"
+
+	"github.com/multiversx/mx-chain-proxy-go/data"
+	"github.com/multiversx/mx-chain-proxy-go/observer/availabilityCommon"
+)
+
+var (
+	errInvalidAvailability           = errors.New("invalid data availability type")
+	errNumNodesMustBeGreaterThanZero = errors.New("the number of nodes must be greater than 0")
+)
+
+// MapCountersHolder handles multiple counters map based on the data availability
+type MapCountersHolder struct {
```

Comment:
can be unexported and use an interface on circularQueueNodesProvider

Reply:
ok, done